gencode.c

```diff
@@ -1552,7 +1585,7 @@ gen_load_macplrel(offset, size)
 		 * as an offset.
 		 */
 		s2 = new_stmt(BPF_LD|BPF_IND|size);
-		s2->s.k = offset;
+		s2->s.k = off_macpl + offset;
```

Comment:
It's not hard to move it to a separate variable, it just wasn't clear to me whether that was actually more robust.